src/OpenLoco/src/Input/MouseInput.cpp

```diff
@@ -335,15 +328,15 @@ namespace OpenLoco::Input
 
             resetMapSelectionFlag(MapSelectionFlags::enable | MapSelectionFlags::enableConstruct | MapSelectionFlags::enableConstructionArrow | MapSelectionFlags::unk_03 | MapSelectionFlags::unk_04);
 
-            if (_toolWidgetIndex >= 0)
+            if (ToolManager::getToolWidgetIndex() >= 0)
```

Comment:
This needs to change into a check for `!= 0xFFFF`

Reply:
I changed the type to int16_t, so this should be fine.